### ISSUE TITLE
Issue with "Move" Operation

### ISSUE THREAD

**Issue by naronatrhea** [RHEAGROUP/CDP4-SDK-Community-Edition#27]
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am running the latest version of the CDP4-SDK
- [x] I have searched [open](https://github.com/RHEAGROUP/CDP4-SDK-Community-Edition/issues) and [closed](https://github.com/RHEAGROUP/CDP4-SDK-Community-Edition/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description
Moving an ordered item does not seem to work in the IME. There is an issue either in the SDK that builds the POST message or in the IME. It seems that the M property of an OrderedItem
 is not set. 

### Steps to Reproduce

This can be for example tested when re-ordering PossibleFiniteState in a PossibleFiniteStateList or re-ordering the PossibleFiniteStateList of an ActualFiniteStateList

### System Configuration

- CDP4 version:
  - [ ] CDP4Common:         
  - [ ] CDP4JsonSerializer: 
  - [ ] CDP4Dal:            
  - [ ] CDP4JsonFileDal:    
  - [ ] CDP4ServicesDal:    
  - [ ] CDP4WspDal:         
  - [x] Other: CDP4-CE     1.0.15.0          
- Environment (Operating system, version and so on):
- .NET Framework version:
- Additional information:

**Comment by samatrhea** [comment]
using this version of the CDP4-IME i created a new EngineeringModel in which i added 1 PossibleFiniteStateList with 3 PossibleFiniteStates. After successful creation i edited this list by reordering one state, i moved the first state to the second position. The following POST message was created:

```
{
	"_delete": [],
	"_create": [],
	"_update": [{
		"classKind": "PossibleFiniteStateList",
		"iid": "12dcad35-cb9c-4759-a688-027ca00de534",
		"modifiedOn": "2018-08-12T14:03:51.652Z",
		"possibleState": [{
			"k": -6835798,
			"m": 108688751,
			"v": "68fbdb31-b208-404f-be16-867eda11a766"
		},
		{
			"k": 108688751,
			"m": -6835798,
			"v": "7b924d5c-8e34-47aa-8bf1-c1a85c9515c7"
		}]
	}],
	"_copy": []
}
```

In this case it seems the move keys were created.

**Comment by samatrhea** [comment]
Another test on a DerivedQuantityKind

1. Create a new DerivedQuantityKind with 2 factors (factors are ordered)

```
{
	"_delete": [],
	"_create": [{
		"alias": [],
		"category": [],
		"classKind": "DerivedQuantityKind",
		"defaultScale": "56863161-fa3a-47b8-a3cd-16465f734b27",
		"definition": [],
		"excludedDomain": [],
		"excludedPerson": [],
		"hyperLink": [],
		"iid": "08c130ca-a270-408f-8069-b7e1d00659f0",
		"isDeprecated": false,
		"modifiedOn": "2018-08-12T14:10:55.560Z",
		"name": "derivedQuantityKind",
		"possibleScale": ["56863161-fa3a-47b8-a3cd-16465f734b27"],
		"quantityDimensionSymbol": "dqk",
		"quantityKindFactor": [{
			"k": -12551680,
			"v": "11aa4db6-40ef-4368-92ac-bda8af9b27a9"
		},
		{
			"k": 77606679,
			"v": "125abfbd-f717-42c0-91da-738316159ea5"
		}],
		"revisionNumber": 0,
		"shortName": "derivedQuantityKind",
		"symbol": "derivedQuantityKind"
	},
	{
		"classKind": "QuantityKindFactor",
		"excludedDomain": [],
		"excludedPerson": [],
		"exponent": "1",
		"iid": "11aa4db6-40ef-4368-92ac-bda8af9b27a9",
		"modifiedOn": "2018-08-12T14:10:45.476Z",
		"quantityKind": "9513aca9-5397-48a1-8187-1683490bdb4f",
		"revisionNumber": 0
	},
	{
		"classKind": "QuantityKindFactor",
		"excludedDomain": [],
		"excludedPerson": [],
		"exponent": "2",
		"iid": "125abfbd-f717-42c0-91da-738316159ea5",
		"modifiedOn": "2018-08-12T14:10:50.681Z",
		"quantityKind": "1c53f8f4-024b-4f42-987a-439901a4bcb2",
		"revisionNumber": 0
	}],
	"_update": [{
		"classKind": "SiteReferenceDataLibrary",
		"iid": "bff9f871-3b7f-4e57-ac82-5ab499f9baf5",
		"parameterType": ["08c130ca-a270-408f-8069-b7e1d00659f0"]
	}],
	"_copy": []
}

```

Step 2: change the order of the factors:

```
{
	"_delete": [],
	"_create": [],
	"_update": [{
		"classKind": "DerivedQuantityKind",
		"iid": "08c130ca-a270-408f-8069-b7e1d00659f0",
		"modifiedOn": "2018-08-12T14:11:56.580Z",
		"quantityKindFactor": [{
			"k": -12551680,
			"m": 77606679,
			"v": "11aa4db6-40ef-4368-92ac-bda8af9b27a9"
		},
		{
			"k": 77606679,
			"m": -12551680,
			"v": "125abfbd-f717-42c0-91da-738316159ea5"
		}]
	}],
	"_copy": []
}
```

again, the move keys seem to be created. I cannot seem to reproduce this issue

**Comment by samatrhea** [comment]
After closing and re-opening the data-source, and opening the EngineeringModel and RDL where these items were added and subsequently re-ordered it seems that the re-order did not happen. Seems to be an issue on the server?

**Comment by samatrhea** [comment]
@naronatrhea diff up at http://phabricator.rheagroup.com/D1403

**Comment by samatrhea** [comment]
fixed in 8f1f2185cc2afb086ad1d5b91f5fa2d2e1e6b051
